### DISCUSSION TITLE
Would like to use sphinx_view in a dedicated app, but struggle with including the url patterns...

### DISCUSSION THREAD

**Discussion by tombreit** [carltongibson/django-sphinx-view#21] [Q&A] [answered]
When trying to reorganize a project structure, I did not manage to include the urlpatterns needed by/for `sphinx_view`. 

### Works

Including the urlpattern (with the string `docs` attached) from in the django project level `urls.py` works fine: 

```python
    path(
        'docs<path:path>',
        DocumentationView.as_view(
            json_build_dir=Path('docs/_build/json/'),
            base_template_name="sphinx_view/base.html",
        ),
        name="documentation",
    ),
```

### Does not work

I tried something like: 

```python
# project/urls.py
urlpatterns = [
    path('admin/', admin.site.urls),
    ...
    path('docs/', include('myapp.urls')),
    ...
]
```

```python
# myapp/urls.py
urlpatterns = [
    path(
        '<path:path>',
        DocumentationView.as_view(
            json_build_dir=Path('docs/_build/json/'),
            base_template_name="sphinx_view/base.html",
        ),
        name="documentation",
    ),
]
```

...but with this setup, it's not possible to match the `docs` route:  

http://fqdn/docs/ →
```
Page not found (404)
Request Method: 	GET
Request URL: 	http://127.0.0.1:8000/docs/

Using the URLconf defined in project.urls, Django tried these URL patterns, in this order:
    admin/
    docs/ <path:path> [name='documentation']
The current path, docs/, didn’t match any of these.
```

http://fqdn/docs/foo/ →
```
Page not found (404)
Request Method: 	GET
Request URL: 	http://127.0.0.1:8000/docs/foo/
Raised by: 	sphinx_view.views.DocumentationView

Using the URLconf defined in project.urls, Django tried these URL patterns, in this order:
    admin/
    docs/ <path:path> [name='documentation']
The current path, docs/foo/, matched the last one.
```

_Any tips on how I can improve this?_

**Comment by carltongibson** [comment, marked as answer]
The path converter swallows the trailing `/` so you'll need to match something like `docs` (without it) when doing the `include`...

Does `docs//` match? 🤔 (or does the double slash get swallowed?)

**Comment by tombreit** [comment]
Good catch - yes, omitting the trailing slash in project.urls.py does the trick!

But in many documentaries I have stumbled across [1] the include-pattern includes the trailing slash. Perhaps sphinx_view would profit from a setup which could handle the trailing slash?

[1] https://docs.djangoproject.com/en/4.0/topics/http/urls/#including-other-urlconfs
